PiwikTracker/PiwikTracker.swift

```diff
@@ -96,6 +96,7 @@ final public class PiwikTracker: NSObject {
             guard events.count > 0 else {
                 // there are no more events queued, finish dispatching
                 self.isDispatching = false
+                self.startDispatchTimer()
```

Comment:
`startDispatchTimer` is called from different levels/abstractions in the code.
Eventually it might become hard to make sure all code branches are covered since the call is spread out in the code.

One option would be if `dispatchBatch()` took a closure that is called when it is done. Then the logic can be confined to the `dispatch()` method (I think).